### ISSUE TITLE
Suggestion: Terrain multi height map layers by terrain patch

### ISSUE THREAD

**Issue by RuanLucasGD** [FlaxEngine/FlaxEngine#2219]
**Issue description:**

The current terrain implementation is good for simple terrains but is a bit limited for more advanced levels. My suggestion to make the terrain more flexible is use different height map layer combined to create the terrain.

Use only one height map is limited because you can't add details without lose the original data, so if you want (for example) create a tool for roads, the road will modify the terrain height-map texture while you are creating the road (this will make a lot of problems when you need modify the road again, because the original terrain was lost).

If you use multi-layers for each terrain detail, like (base height, roads, holes...) you can make all separated without lose the terrain base, and this help to create another tools for procedural terrains for example.

**Current Height Map Implementation:**

![image](https://github.com/FlaxEngine/FlaxEngine/assets/79365912/bcc928dd-7ba9-40da-b51f-d15acf9f8561)

**Suggested Height Map Implementation:**

![image](https://github.com/FlaxEngine/FlaxEngine/assets/79365912/9c74ec07-681b-4053-b988-30574f30f4bc)

Basically, each layer is an texture, blending all layers we will have an main height map texture. Maybe would be good an way to add and remove the layers dynamically. All layers are assigned on the terrain patch, like the currently implementation.

Each layer will sun with previous layer, like it: layer1 + layer2 + layer3....

**Comment by AlTheSlacker** [comment]
I wrote some code a while back to bring procedural heightmaps into the Flax editor: https://github.com/AlTheSlacker/FlaxProcTerrain

You will see that I load all of the patches into one large array and then iterate noise onto this array before splitting it back into the original terrain patches. You could save the state of the master array to provide the incremental approach that you talk about (addition of multiple master arrays would be straightforward), although it would probably be best to save the states to their own file as large terrains would generate a lot of data. I'm wrapped up in other projects now, but feel free to help yourself to any of my code.
